.github/workflows/CI.yml

```diff
@@ -0,0 +1,425 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+  pull_request:
+  merge_group:
+  workflow_dispatch:
+    inputs:
+      test_release:
+        description: If true, publish to test.pypi.org
+        required: true
+        default: true
+        type: boolean
+
+permissions:
+  contents: read
+
+env:
+  DOCKER_IMAGE: ghcr.io/onekey-sec/unblob
+
+jobs:
+  check_pre_commit:
+    name: Check - pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+
+      - name: Setup 3rd party dependencies
+        uses: ./.github/actions/setup-dependencies
+
+      - name: Setup Nix
+        uses: cachix/install-nix-action@v30
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.18.8/install
+
+      - name: Check pre-commit hook
+        uses: pre-commit/action@v3.0.1
+
+  check_pyright:
+    name: Check - pyright
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Setup 3rd party dependencies
+        uses: ./.github/actions/setup-dependencies
+
+      - name: Check - pyright
+        run: uv run pyright .
+
+  run_python_tests:
+    name: Run tests (Python)
+    needs: [check_pre_commit, check_pyright]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v4
+
+      - name: Setup 3rd party dependencies
+        uses: ./.github/actions/setup-dependencies
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Setup git lfs
+        uses: ./.github/actions/setup-git-lfs
+
+      - name: Run pytest
+        run: uv run pytest -vvv
+
+  build_linux_wheels:
+    name: Build wheels (linux)
+    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch' || contains(github.event.*.labels.*.name, 'dependencies')
+    needs: [check_pre_commit]
+    runs-on: ubuntu-24.04
```

Comment:
We use `ubuntu-latest` elsewhere but we use `ubuntu-24.04` here. We should standardize and use the same version everywhere.